services/src/integ-test/java/uk/gov/moj/sdt/services/messaging/IndividualRequestMdbIntTest.java

```diff
@@ -30,76 +30,103 @@
 import uk.gov.moj.sdt.test.utils.TestConfig;
 
-import java.io.IOException;
+import javax.jms.JMSException;
+import javax.jms.ObjectMessage;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 /**
  * IntegrationTest class for testing the MessageReader implementation.
  *
  * @author Manoj Kulkarni
  */
 @ActiveProfiles("integ")
-@RunWith(SpringRunner.class)
-@SpringBootTest(classes = {TestConfig.class, ServicesTestConfig.class })
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {TestConfig.class, ServicesTestConfig.class})
 @Sql(scripts = {"classpath:uk/gov/moj/sdt/services/sql/RefData.sql", "classpath:uk/gov/moj/sdt/services/sql/IndividualRequestMdbIntTest.sql"})
+@Transactional
 public class IndividualRequestMdbIntTest extends AbstractIntegrationTest {
+
+    @Autowired
+    @Qualifier("MessageWriter")
+    private MessageWriter messageWriter;
+
+    @Autowired
+    @Qualifier("messageListenerContainer")
+    private DefaultMessageListenerContainer mcolsContainer;
+
+    @Autowired
+    @Qualifier("messageListenerContainerMCol")
+    private DefaultMessageListenerContainer mcolContainer;
+
+    @MockBean
+    private ITargetApplicationSubmissionService targetApplicationSubmissionService;
+
+    private IndividualRequestMdb individualRequestMdb;
+
+    private SdtMessage sdtMessage;
+
     /**
      * Setup the test.
      */
-    @Before
+    @BeforeEach
     public void setUp() {
+        individualRequestMdb = new IndividualRequestMdb(targetApplicationSubmissionService);
         // Write a Message to the MDB
-        final ISdtMessage sdtMessage = new SdtMessage();
-        sdtMessage.setSdtRequestReference("SDT_REQ_TEST_1");
-        sdtMessage.setMessageSentTimestamp(System.currentTimeMillis());
-        sdtMessage.setEnqueueLoggingId(1);
-        final IMessageWriter messageWriter = (IMessageWriter) this.applicationContext.getBean("MessageWriter");
+        sdtMessage = createSdtMessage();
+
         messageWriter.queueMessage(sdtMessage, "MCOLS", false);
     }
 
     /**
      * This method tests the read message.
      *
      * @throws InterruptedException if the thread call is interrupted
-     * @throws IOException          if there is any problem when reading the file
+     * @throws JMSException         if there is any problem when reading the file
      */
     @Test
-    public void testReadMessage() throws InterruptedException, IOException {
+    public void testReadMessage() throws InterruptedException, JMSException {
         Thread.sleep(5000);
-        Assert.assertTrue("Submission read successfully.", true);
+        ObjectMessage objectMessage = mock(ObjectMessage.class);
+        when(objectMessage.getObject()).thenReturn(sdtMessage);
+        individualRequestMdb.readMessage(objectMessage);
+        assertTrue(true, "Submission read successfully.");
     }
```

Comment:
It appears that this method originally tested `IndividualRequestMdb.readMessage()` indirectly.  It did this by calling `MessageWriter.queueMessage()` in `setUp()`, then waiting 5 seconds to ensure that `IndividualRequestMdb` had been called by it.  The call to `MessageWriter` is still in `setUp()`, so by adding the direct call to `IndividualRequestMdb.readMessage()` here it's now being called twice.

It's probably best to change the test so that `IndividualRequestMdb.readMessage()` is only called once.  This can be done by either removing the direct call here or by removing the indirect call from `setUp()`.

Whichever option is chosen, add the following Mockito verify line to confirm that the `TargetApplicationSubmissionService.processRequestToSubmit()` method has been called once.

`verify(targetApplicationSubmissionService).processRequestToSubmit("SDT_REQ_TEST_1", null); `

Reply:
addressed